Clean area/connectors related github issues (#1929)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/applications-databases-mssql.md

id: applications-databases-mssql
title: Microsoft SQL Server
---
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Pack Assets

### Templates

The Centreon Plugin Pack **Microsoft SQL Server** brings a host template:

* App-DB-MSSQL-custom

It brings the following service templates:

| Service Alias        | Service Template                  | Service Description                                                                                                           | Default |
|:---------------------|:----------------------------------|:------------------------------------------------------------------------------------------------------------------------------|:--------|
| Backup-Age           | App-DB-MSSQL-Backup-Age           | Check database backups of the server                                                                                          |         |

> | Blocked-Processes    | App-DB-MSSQL-Blocked-Processes    | Check blocked processes on the server. Service cannot work because of a SQL request. It depends of your MS SQL Server version | X       |

Reviewer: ?

| Connection-Time      | App-DB-MSSQL-Connection-Time      | Check the connection time to the server. This time is given in seconds                                                        | X       |
| Databases-Size       | App-DB-MSSQL-Databases-Size       | Check database data and logs files                                                                                            | X       |
| Deadlocks            | App-DB-MSSQL-Deadlocks            | Check deadlocks per second of the server                                                                                      | X       |
| Failed-Jobs          | App-DB-MSSQL-Failed-Jobs          | Check MSSQL failed jobs                                                                                                       | X       |
| Locks-Waits          | App-DB-MSSQL-Locks-Waits          | Check locks-waits per second of the server                                                                                    |         |
| Page-Life-Expectancy | App-DB-MSSQL-Page-Life-Expectancy | Check the "Page Life Expectancy" of the server. No alerts by default                                                          |         |

> | Sql-Statement        | App-DB-MSSQL-Sql-Statement        | Check allowing to execute a custom SQL request with a digital answer                                                          |         |

Reviewer: digital?

> | Sql-Statement-String | App-DB-MSSQL-Sql-Statement-String | Check allowing to execute a custom SQL request with a string answer                                                           |         |

Reviewer: ```suggestion
| Sql-Statement-String | App-DB-MSSQL-Sql-Statement-String | Check allowing to execute a custom SQL query with a string answer                                                           |         |
```

| Metric Name            | Unit  |
|:-----------------------|:------|
| mssql.lockswaits.count | count |

</TabItem>
<TabItem value="Page-Life-Expectancy" label="Page-Life-Expectancy">

| Metric Name                  | Unit  |
|:-----------------------------|:------|
| page.life.expectancy.seconds | s     |

</TabItem>
<TabItem value="Sql-Statement" label="Sql-Statement">

| Metric Name                       | Unit  |
|:----------------------------------|:------|
| sqlrequest.execution.time.seconds | s     |
| value                             |       |

</TabItem>
<TabItem value="Sql-Statement-String" label="Sql-Statement-String">

| Metric Name   | Unit  |
|:--------------|:------|
| *rows*#string |       |

</TabItem>
<TabItem value="Transactions" label="Transactions">

| Metric Name                                | Unit  |
|:-------------------------------------------|:------|
| databases.transactions.persecond           | /s    |
| *database*#database.transactions.persecond | /s    |

</TabItem>
</Tabs>

## Prerequisites

### Monitoring user

To use this probe, the configuration of a monitoring user with specific privileges is required. The `serveradmin` role 

> can be used for testing purpose but should never be used in production because of the security risk involved. 

Reviewer: ```suggestion
can be used for testing purposes but should never be used in production because of the security risk involved. 
```

> Birk Bohne, who is a valuable contributor of the _check\_mssql\_health_ probe wrote a script to assign a least-privilege user. 

Reviewer: ```suggestion
Birk Bohne, who is a valuable contributor of the _check\_mssql\_health_ probe, wrote a script to assign a less-privileged user. 
```

His script can be found under the **Preparation of the database** section of the following [documentation](https://github.com/lausser/check_mssql_health/blob/master/doc/check_mssql_health.en.txt). 

> We recommend you to use a domain user to better manage its properties and privileges. 

Reviewer: ```suggestion
We recommend you use a domain user to better manage its properties and privileges. 
```

### Dependencies

> Those packages are required: `freetds perl-DBD-Sybase unixODBC`

Reviewer: ```suggestion
These packages are required: `freetds perl-DBD-Sybase unixODBC`
```

```bash
dnf install centreon-plugin-Applications-Databases-Mssql
```

</TabItem>
<TabItem value="CentOS 7" label="CentOS 7">

```bash
yum install centreon-plugin-Applications-Databases-Mssql
```

</TabItem>
<TabItem value="Debian 11" label="Debian 11">

```bash
apt install centreon-plugin-applications-databases-mssql
```

</TabItem>
</Tabs>

## Configuration

### Host

* Log into Centreon and add a new host through **Configuration > Hosts**.
* Fill the **Name**, **Alias** & **IP Address/DNS** fields according to your **SQL Server database** server settings.
* Apply the **App-DB-MSSQL-custom** template to the host.
* Once the template is applied, fill in the corresponding macros. Some macros are mandatory.

| Mandatory   | Macro         | Description                                                                            |
|:------------|:--------------|:---------------------------------------------------------------------------------------|
|             | EXTRAOPTIONS  | Any extra option you may want to add to every command line (eg. a --verbose flag)      |
|      X      | MSSQLPASSWORD | Monitoring user password (Default: 'PASSWORD')                                         |

> |      X      | MSSQLPORT     | MSSQL Instance listening port (Default: '1433')                                        |

Reviewer: ```suggestion
|      X      | MSSQLPORT     | MSSQL instance listening port (Default: '1433')                                        |
```
